### PR TITLE
Fix test failure and warnings on Elixir 1.7

### LINKED ISSUE #375: Remove usage of `System.stacktrace` outside of `rescue/catch` clauses
```
warning: System.stacktrace/0 outside of rescue/catch clauses is deprecated. If you want to support only Elixir v1.7+, you must access __STACKTRACE__ inside a rescue/catch. If you want to support earlier Elixir versions, move System.stacktrace/0 inside a rescue/catch.
```

### LINKED ISSUE #376: Fix test failure in ErrorMatcherTest on Elixir 1.7
```
  1) test Task (Appsignal.ErrorHandler.ErrorMatcherTest)
     test/appsignal/error_handler/error_matcher_test.exs:193
     Assertion with =~ failed
     code:  assert message =~ expected
     left:  "Process #PID<0.2255.0> terminating"
     right: ~r/Process #PID<[\d.]+> terminating: {:function_clause, \[{Float, :cei.../
     stacktrace:
       test/appsignal/error_handler/error_matcher_test.exs:248: Appsignal.ErrorHandler.ErrorMatcherTest.message/2
       test/appsignal/error_handler/error_matcher_test.exs:199: (test)
```

### PR DESCRIPTION
**Pull request by jeffkreeftmeijer**
Fixes #376.
Fixes #375.

This patch adds an exception in the error matcher test to fix a failure on 1.7 in combination with OTP 20 and up. The code works, but the reported error is different, which caused a test to fail. We're revisiting the error matcher in #357.

1.7 throws warnings when calling System.stacktrace/0 outside of a `try`/`catch` block. Instead of falling back to `System.stacktrace`, we now explicitly pass the stacktace internally. `Appsignal.send_error/1-7` had the stack trace fallback too. Instead, this patch makes it fall back to an empty list with a deprecation warning.

### CODE REVIEW

File: lib/appsignal/demo.ex [hunk 1]
```diff
@@ -15,7 +15,7 @@ defmodule Appsignal.Demo do
     |> Appsignal.Transaction.set_error(
       "TestError",
       "Hello world! This is an error used for demonstration purposes.",
-      System.stacktrace
+      []
```
Comment (tombruijn) [inline lib/appsignal/demo.ex]: this makes the demo error page quite plain. Could we raise an error in this function and catch it to set the stacktrace?

Comment (jeffkreeftmeijer) [reply lib/appsignal/demo.ex]: Absolutely! https://github.com/appsignal/appsignal-elixir/pull/386/commits/98bbaf03245b5d7d7e3dde9f654dd981f2a0adf9

File: .travis.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ elixir:
   - 1.4.5
   - 1.5.3
   - 1.6.6
+  - 1.7.0-rc.0
```
Comment (tombruijn) [inline .travis.yml]: I saw rc 1 came out yesterday: https://elixirforum.com/t/elixir-v1-7-0-rc-1-is-out/15221
Should we update it to test against that?

**Comment by tombruijn** [comment]
> 1.7 throws warnings when calling System.stacktrace/0 outside of a try/catch block. Instead of falling back to System.stacktrace, we now explicitly pass the stacktace internally. 

Should all warnings be fixed? Because I can still some in the test suite on the CI: https://travis-ci.org/appsignal/appsignal-elixir/jobs/406145170

**Comment by tombruijn** [comment]
As discussed, the warnings I mentioned in my previous comment only occur in the test suite. So we can merge it.

File: .travis.yml (hunk 2 above)
Comment (tombruijn) [reply .travis.yml]: Could not get it to work on 1.7.0.rc.1. Something about a missing package for OTP 19: https://travis-ci.org/appsignal/appsignal-elixir/jobs/407046279

```
Installing Elixir 1.7.0-rc.1
0.05s$ wget https://repo.hex.pm/builds/elixir/v1.7.0-rc.1-otp-19.zip
--2018-07-23 09:06:28--  https://repo.hex.pm/builds/elixir/v1.7.0-rc.1-otp-19.zip
Resolving repo.hex.pm (repo.hex.pm)... 151.101.22.2, 2a04:4e42:5::514
Connecting to repo.hex.pm (repo.hex.pm)|151.101.22.2|:443... connected.
HTTP request sent, awaiting response... 403 Forbidden
2018-07-23 09:06:28 ERROR 403: Forbidden.
The command "wget https://repo.hex.pm/builds/elixir/v1.7.0-rc.1-otp-19.zip" failed and exited with 8 during .
```

**Pull request merged**